auction-backend/src/main/java/com/atlantbh/internship/auction/app/model/impl/JaccardIndex.java

```diff
@@ -0,0 +1,88 @@
+package com.atlantbh.internship.auction.app.model.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class JaccardIndex {
+    final static int OR_BY = 32; // ASCII difference between uppercase and lowercase letters
+    final static double MINIMUM_ALLOWED_SIMILARITY = 0.3;
+    Map<Character, Integer> patternCharFrequency;
+
+    public JaccardIndex(final String pattern) {
+        patternCharFrequency = calculateCharacterFrequency(pattern);
+    }
```

Comment:
It is not a best practice to have a method call in the constructor.
Does it make more sense for this class to be static?